scheduler/scheduler.go

```diff
@@ -573,38 +583,42 @@ func (s *scheduler) stopTask(id, source string) []serror.SnapError {
 			serror.New(ErrTaskDisabledNotStoppable),
 		}
 	default:
-		// Group depndencies by the host they live on and
-		// unsubscirbe them since task is stopping.
-		depGroupMap := s.gatherMetricsAndPlugins(t.workflow)
-
+		// Group dependencies by the host they live on and
+		// unsubscribe them since task is stopping.
+		depGroupMap := make(map[string]depGroup)
+		depGroupMap[""] = depGroup{Metrics: []core.Metric{}}
+		getWorkflowPlugins(t.workflow.processNodes, t.workflow.publishNodes, depGroupMap)
 		var errs []serror.SnapError
 		for k := range depGroupMap {
 			mgr, err := t.RemoteManagers.Get(k)
 			if err != nil {
 				errs = append(errs, serror.New(err))
 			} else {
-				uerrs := mgr.UnsubscribeDeps(t.ID(), depGroupMap[k].Metrics, returnCorePlugin(depGroupMap[k].Plugins))
+				// empty metric list is passed to inform UnsubscribeDeps is called from scheduler on stop task
+				uerrs := mgr.UnsubscribeDeps(t.ID(), []core.Metric{}, returnCorePlugin(depGroupMap[k].Plugins))
```

Comment:
Same as above.   Consider refactoring the exported `UnsubscribeDeps` so it only takes taskid and  []plugins.